flocker/node/agents/gce.py

```diff
@@ -35,6 +33,25 @@
 _METADATA_SERVER = u'http://169.254.169.254/computeMetadata/v1/'
 _METADATA_HEADERS = {u'Metadata-Flavor': u'Google'}
 
+# timeouts were gathered by running each operation 250 times. The
+# chosen timeouts are 3-5 times higher than the maximum time taken for
+# each operation.  This should be a couple of standard deviations from
+# the mean.
+# All this might fall apart if GCE is having a very slow day...
+VOLUME_DEFAULT_TIMEOUT = 120
+VOLUME_LIST_TIMEOUT = 10
+VOLUME_DELETE_TIMEOUT = 20
+VOLUME_INSERT_TIMEOUT = 20
+VOLUME_ATTACH_TIMEOUT = 90
+VOLUME_DETATCH_TIMEOUT = 120
+
+
+class GCEVolumeException(Exception):
```

Comment:
I'm unsure if I should create a new exception type for every case where GCE can throw an interesting error (cannot destroy volume when attached to an instance, dataset already exists, etc.)